### ISSUE TITLE
Replace `Context::notify_interval` with `SendFuture::repeat_every`

### ISSUE THREAD

**Issue by thomaseizinger** [Restioson/xtra#125]
Extracted out of #71.

I am proposing to implement this as a first step as-is in the current library. Perhaps we can later extract it into an extension crate.

Important aspects to cover:

- Must work only on `Handler`s that return `()`
- Must be feature-flagged on `timing`

**Comment by thomaseizinger** [comment]
Fair point. Would you be okay with removing the feature entirely and adding it back later once the dust around the slimming down has settled?

**Comment by Restioson** [comment]
Sure
